pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -283,6 +288,10 @@ func (g *DefaultValidator) validateTrialTemplate(instance *experimentsv1beta1.Ex
 		if _, ok := trialParametersRefs[parameter.Reference]; ok {
 			return fmt.Errorf("parameter reference %v can't be duplicated in spec.trialTemplate.trialParameters: %v", parameter.Reference, trialTemplate.TrialParameters)
 		}
+		// Check if parameter references exist in experiment parameters
+		if _, ok := experimentParameterNames[parameter.Reference]; !ok {
+			return fmt.Errorf("parameter reference %v does not exists in spec.Parameters: %v", parameter.Reference, instance.Spec.Parameters)
+		}
```

Comment:
IMHO, How about move these codes after
```go
trialParametersNames[parameter.Name] = true
trialParametersRefs[parameter.Reference] = true
```

It may not important, but I feel like these should be right after codes right after checking duplicates.

Reply:
Thanks for the suggestion, done :)